### ISSUE TITLE
Getting the gradient of the gradient

### ISSUE THREAD

**Issue by zanellar** [pytorch/pytorch#103509]
I have a function f(x) which includes an approximation with Taylor expansion, e.g. f(x,a)=g(a)+g’(a)(x-a) where g’(x)=dg(x)/dx is computed with torch.autograd. Now I would like to differentiate it in “a”, i.e. df(x,a)/da. To simplify even more the function let’s say that f(a)=g’(a):

```
def g(x):
    return x

def f(a):
    return torch.autograd.functional.jacobian(g, a)
```

However, if I visualize the computational graph, it seems that I lose the gradient when computing the Jacobian. In fact, using the following lines I get an empty graph:
```
a = torch.ones(1, requires_grad=True)
torchviz.make_dot(f(a), params=dict(a=a))
```
At this point, I wonder if there is a way to implement this computation 

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by Gabrui** [comment]
You should use `grad` with the additional argument of `create_graph=True`. I was doing a similar thing, I saw an example in https://nn.labml.ai/gan/wasserstein/gradient_penalty/index.html

Just read the docs, for the jacobian, you can also use  create_graph=True

**Comment by Gabrui** [comment]
Oh hey, no problem. I was just reading what I wrote, and I am sorry if I sounded rude. What I meant was that I had just read the docs. I am working in a similar problem in which I need to optimize over the gradients. Also I am studying system identification problems and trying to write it in pytorch. You can reach me via: gam at ita dot br
